corehq/apps/receiverwrapper/tests/test_submissions.py

```diff
@@ -405,3 +406,48 @@ def test_no_rate_limiting(self, allow_usage):
         form_xml = self.get_xml('simple_form')
         submit_form_locally(form_xml, domain=self.domain, max_wait=None)
         allow_usage.assert_not_called()
+
+
+class SubmitFormLocallyTest(TestCase, TestFileMixin):
+    root = os.path.dirname(__file__)
+    file_path = ('data',)
+
+    @classmethod
+    def setUpClass(cls):
+        super(SubmitFormLocallyTest, cls).setUpClass()
+        cls.domain = 'atestdomain'
+        cls.project = create_domain(name=cls.domain)
+
+        cls.app = Application(domain=cls.domain, version=4)
+        cls.app.save()
+        cls.app_id = cls.app.get_id
+
+        cls.build = Application(domain=cls.domain, version=3)
+        cls.build.copy_of = cls.app_id
+        cls.build.save()
+        cls.build_id = cls.build.get_id
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.project.delete()
```

Comment:
`cls.app` and `cls.build` should be cleaned up too, right?

Reply:
Domain deletion should handle that. The tests I used as an example did teardown like this, but Idid verify that it happens [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/models.py#L812-L813)). Happy to make it more explicit if that's better